trolldb/config/config.py

```diff
@@ -108,37 +109,38 @@ class AppConfig(BaseModel):
     subscriber: SubscriberConfig
 
 
-@validate_call
-def parse_config_yaml_file(filename: FilePath) -> AppConfig:
-    """Parses and validates the configurations from a YAML file.
+@logger.catch(onerror=lambda _: sys.exit(1))
+def parse_config(file: int | str | bytes | PathLike[str] | PathLike[bytes]) -> AppConfig:
```

Comment:
Do we really need the type hinting here? If python decides to support a new type for opening, our code will need to be changed for this new functionality to be reflected here, so we are not forward-compatible.

Reply:
Done. The file type hint has been removed.